ls1intum/Artemis
# `Communication`: Show message icon when users have unread messages

### Checklist
#### General



#### Client

### Motivation and Context


Users can only be aware of unread messages when they go to conversations.  It would be nice to have a icon showing up in Messages tab while users are in any other tab such as Exercises, Lectures, Statistics.
### Description


Currently, `MetisConversationService` is being setup when Messages tab is open which means, only after `course-conversations.component.ts` is initialized. 

In this PR `MetisConversationService` is setup when `course-overview.component.ts` is initialized which allows now to access conversations from course-overview.component.ts and check if there are unread messages.  Also a new Subject is introduced at `MetisConversationService` called hasUnreadMessages, where now we subscribe to it and get the latest info.

### Steps for Testing

Prerequisites:
- 2 Users registered at the same course
- 1 Course

1. Log in to Artemis
2. Navigate to Course Messages tab 
3. Search for the user you want to send the message and then send it
4. Login as the other user and navigate to course overview 
5. Then check if you got the Message icon in the Messages tab 

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-overview.component.ts | 76.87% | 92.3% |
| course-conversation.component.ts | 100% | 100% |
| metis-conversation.service.ts | 70.7% | 84.18% |


### Screenshots





https://user-images.githubusercontent.com/26312757/215340045-70d15359-fbb3-446d-95de-1f96a7b0d61f.mp4
